economic_analysis/Facades Economic Analysis.py

```diff
@@ -0,0 +1,155 @@
+
+#define constants
+btu_propane_in_one_gallon = 91452 #source: https://www.eia.gov/energyexplained/units-and-calculators/
+gallons_to_barrels = 42 #source: https://www.eia.gov/energyexplained/units-and-calculators/
+btu_fueloil_in_one_barrel = 6287000 #source: https://www.eia.gov/energyexplained/units-and-calculators/
+
+
+##define functions
+
+#choose results columns to work with
+selected_cols = [
+    'building_id', 
+    'simulation_output_report.total_site_electricity_kwh', 
+    'simulation_output_report.total_site_natural_gas_therm',
+    'simulation_output_report.total_site_fuel_oil_mbtu',
+    'simulation_output_report.total_site_propane_mbtu',
+    'simulation_output_report.total_site_energy_mbtu',
+    'simulation_output_report.upgrade_cost_usd']
+def downselect_cols(df):
+	df = df[selected_cols]
+	return df
+
+#calculate total energy difference for each upgrade, for each home, add to the mini dataframes
+def calc_delta_elec(df_upgrade, df_baseline):
+	return (df_upgrade['simulation_output_report.total_site_electricity_kwh']-df_baseline['simulation_output_report.total_site_electricity_kwh'])
+def calc_delta_ng(df_upgrade, df_baseline):
+	return (df_upgrade['simulation_output_report.total_site_natural_gas_therm']-df_baseline['simulation_output_report.total_site_natural_gas_therm'])
+def calc_delta_fo(df_upgrade, df_baseline):
+	return (df_upgrade['simulation_output_report.total_site_fuel_oil_mbtu']-df_baseline['simulation_output_report.total_site_fuel_oil_mbtu'])
+def calc_delta_lp(df_upgrade, df_baseline):
+	return (df_upgrade['simulation_output_report.total_site_propane_mbtu']-df_baseline['simulation_output_report.total_site_propane_mbtu'])
+def calc_delta_energy(df_upgrade, df_baseline):
+    return(df_upgrade['simulation_output_report.total_site_energy_mbtu']-df_baseline['simulation_output_report.total_site_energy_mbtu'])
+
+#calculate year1 utility bill savings
+def calc_year1_bill_savings(var_util_costs, delta_energy):
+	return (var_util_costs*delta_energy)
+def calc_year1_fo_bill_savings(var_util_costs, delta_energy):
+	return (delta_energy *
+            var_util_costs*
+            gallons_to_barrels*
+            (1/btu_fueloil_in_one_barrel)*
+            1000000)
+def calc_year1_lp_bill_savings(var_util_costs, delta_energy):
+    return(delta_energy *
+           var_util_costs*
+           (1/btu_propane_in_one_gallon)*
+           1000000)
+
+#calculate simple payback period
+def calc_spp(upfront_cost, year1_savings):
+	return(upfront_cost/year1_savings)
+
+#calculate npv
+analysis_period = 30
+discount_rate = 0.034
+def calc_npv(upfront_cost, year1_savings): #note: assuming 30 year lifetime for everything, which is not a general solution to this situation
+	npv_list = []
+	for cost, savings in zip(upfront_cost, year1_savings): #this loops through all the rows in the results_csv (ie each dwelling unit)
+		cost_array = [0]*(analysis_period+1)
+		cost_array[0] = cost
+		savings_array = [savings] * (analysis_period+1)
```

Comment:
@ekpresent Since you are starting the equation at year 0, shouldn't the savings array start with 0? eg: `[0, savings_yr1, savings_yr2, ..., savings_yr_n]` so that you would not have any savings at time 0.

You can update with `savings_array[0] = 0`

Reply:
I think I mimicked how @joseph-robertson  was doing it. If the upgrade happens on 1/1/00, then Year 0 has both the upgrade cost and a year of savings. If it happens on 12/31/00, then Year 0 has the upgrade cost and no savings. Not sure which is better. @ejhw do you have a preference?

Reply:
@ekpresent I think upgrade cost should happen at year 0 (no discounting) and the first year of savings should happen at year 1 (discounted by 1 year).